tests/common/downloader/mod.rs

```diff
@@ -0,0 +1,166 @@
+//!
+//! The compiler downloader.
+//!
+
+pub mod config;
+pub mod solc_list;
+
+use std::os::unix::fs::PermissionsExt;
+use std::path::Path;
+use std::path::PathBuf;
+use std::str::FromStr;
+
+use colored::Colorize;
+
+use self::config::binary::protocol::Protocol;
+use self::config::Config;
+use self::solc_list::SolcList;
+
+///
+/// The compiler downloader.
+///
+#[derive(Debug)]
+pub struct Downloader {
+    /// The `reqwest` HTTP client.
+    http_client: reqwest::blocking::Client,
+    /// The solc-bin JSON list metadata.
+    solc_list: Option<SolcList>,
+}
+
+impl Downloader {
```

Comment:
@hedgar2017 , I've decided to copy implementation of the `Downloader` from the `compiler-tester` here and have a similar `solc-bin.json` config for it.

I was thinking of two other solutions:
1. Greatly simplify `Downloader` to just downloaded hardcoded versions from the tests
  * I was thinking that we probably want to be agile and support the same config as in the tester? 
3. Re-use `Downloader` from the `compiler-tester`.
  * The argument to not go this way was not to have `tester` as a dependency for FE compiler (even for tests)

I'm not certain in this solution, so, need your advice here. What do you think would be the best for us here?

Reply:
Let's convert `era-compiler-solidity` to a workspace and make `downloader` a separate crate.
We'll need the workspace for @sayon's Yul parser anyway.

Reply:
But the problem is that the Downloader is also used for vyper, and we can't have the same code in both crates.
Maybe worth doing it in `era-compiler-common` by converting it into a workspace? The repo was made for common stuff not depending on LLVM.

Reply:
That makes total sense, I'll put downloaded to `era-compiler-common`. Thanks!